pkg/apis/authorization/types.go

```diff
@@ -121,4 +133,8 @@ type SubjectAccessReviewStatus struct {
 	Allowed bool
 	// Reason is optional.  It indicates why a request was allowed or denied.
 	Reason string
+	// EvaluationError is an indication that some error occurred during the authorization check.
+	// It is entirely possible to get an error and be able to continue determine authorization status in spite of it.
+	// For instance, RBAC can be missing a role, but enough roles are still present and bound to reason about the request.
+	EvaluationError string
```

Comment:
This field allows us to cleanly mirror the `Authorize() (allowed bool, reason string, evaluationError error)` signature we have.